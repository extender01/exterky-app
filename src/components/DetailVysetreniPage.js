import React from "react";
import  { connect } from "react-redux";
import { startPridatPoznamkuOdd } from "../actions/vysetreniActions";
import TestPage from "./TestPage";







                        //destructured props
export class DetailVysetreniPage extends React.Component {


  constructor(props) {
    super(props);
    this.state = {
      isPozOddFormVisible: false,   //viditelnost formulare k poznamce
      isPozOddVisible: true,        //viditelnost samotne poznamky
      poznOdd: this.existujePoznOdd() ? this.props.vv.poznamkaOdd[this.login] : ""
      
    };
  };

  login = this.props.uzivatel.substring(0, this.props.uzivatel.indexOf("@")); // cast stringu uzivatele pred @sno.cz


  toggleFormular = (e) => {
      e.preventDefault();
      this.setState(() => ({
        isPozOddFormVisible: !this.state.isPozOddFormVisible,
        isPozOddVisible: !this.state.isPozOddVisible
      }));
  };

  priZmene = (e) => {
    const poznOdd = e.target.value;
    this.setState(() => ({
      poznOdd
    }));
    
  };

  provedSubmit = (e) => {
    e.preventDefault();
    console.log("proveden submit");
    this.props.provedDispatchPridatPoznamkuOdd(this.props.vv.id, {...this.props.prevPoznOdd, [this.login]: this.state.poznOdd});
    this.toggleFormular(e);
    //this.props.history.push("/");
    
          
  };

  //kontrola jestli neco existuje v poznamce od oddeleni, kdyz jo tak dole v JSX se to ternary operatorem evaluuje jestli je to od soucasne prihlaseneho uzivatele
  existujePoznOdd = () => {
     const tempObj = this.props.vv;
     
    if  (tempObj.hasOwnProperty("poznamkaOdd")) {
      //console.log("true")
      return true
    } else {
      //console.log("false")
      return false
    };
   };

   kdyJezdi = () => {
    console.log("ma to byt krnov");
    switch (this.props.vv.kam) {
       case "Krnov":
       console.log("je to krnov");
        return (<p>casy od krnova</p>)
     };
   };



   
  
   typyOdberu = {
     sg: "Srazliva krev s gelem",
     se: "Srazliva krev bez gelu",
     pli: "Nesrazliva s Li heparinem",
     nespec: "oddelovaci",
     specialni: "...ahoj"

     //dopsat Object.keys a z jeho array vytahnout specialni, kde to bude primo napsane
   };


   
   coZaOdber = () => {
  
   
    

    
    let odberPole = this.props.vv.odber.split(", ");
    console.log(odberPole);
   
   

      return (
        odberPole.map((item, index) => {
              return <div key={index} className="blok"><div className={`obrazek ${item}`}></div><p>{this.typyOdberu[item]}</p></div>;
        }))


      
    };

  

  render() {

    const {vv, uzivatel} = this.props;  //desctructuring, abychom nemuseli vsude psat this.props.vv ...
    const {isPozOddFormVisible, isPozOddVisible} = this.state; //zas abysme nemuseli psat this.state.isPozOddVisible
    
    
   

    return (
      <div className="hlavni">
      <div className="flex-container">
          <div className="box grow1 blue">
            <h4>NÁZEV</h4>
            <p>{vv.nazev}</p>
          </div>
          <div className="box  grow2 blue">
            <h4>SYNONYMA</h4>
            <p>{vv.synonyma}</p>
          </div>
      </div>
      <div className="flex-container">
          <div className="box grow1 blue">
            <h4>NÁZEV V AKORDU</h4>
            <p>{vv.nazevAk}</p>
          </div>
          <div className="box  grow2 blue">
            <h4>SKUPINY V AKORDU</h4>
            <p>{vv.skAk}</p>
          </div>
      </div>
      <div className="flex-container">
        <div className="box grow1 vio">
          <h4>KAM TO JEDE</h4>
          <p>{vv.kam}</p>
        </div>
        <div className="box grow2 vio">
          <h4>KDY JEZDÍ</h4>
          {this.kdyJezdi()}
        </div>
      </div>
      <div className="flex-container">
        <div className="box grow1 yel">
          <h4>ODBĚR</h4>
          <div className="lajna">

           {/*} <div className="blok">
              <div className="obrazek serum"></div>
              <p>Srazliva krev s gelem</p>
            </div>
            <div className="blok">
              <div className="obrazek serum2"></div>
              <p>Srazliva krev bez gelu</p>
            </div>
            <div className="blok">
              <div className="obrazek plasma"></div>
              <p>Li heparin nesrazliva</p>
             </div> */}
            {this.coZaOdber(this.props.vv.odber)}
            
          </div>
          
        </div>
        <div className="box grow2 yel">
          <h4>PREANALYTICKÁ FÁZE</h4>
          <p>{vv.preanal}</p>
        </div>
      </div>
      <div className="flex-container">
        <div className="box grow1 yel">
          <h4>POZNÁMKA OBECNÁ</h4>
          <p>{vv.poznamka}</p>
        </div>
        <div className="box grow1 yel">
          <h4>POZNÁMKA ODDĚLENÍ {this.login.toUpperCase()} <span id="pridatPozOdd" onClick={this.toggleFormular}>+</span></h4>
          <div className={isPozOddFormVisible ? "neni-videt ale-je" : "neni-videt"}>
            <form onSubmit={this.provedSubmit}>
              <textarea rows="4" onChange={this.priZmene} value={this.existujePoznOdd() ? this.state.poznOdd : undefined} />
              <button>Pridat</button>
            </form>
          </div>
          
          <div  className={isPozOddVisible ? undefined : "neviditelny"}>
            {this.existujePoznOdd() ? <p> {this.state.poznOdd}</p> : <p>neni tam</p>}
          </div>
        </div>
        
      </div>
      <div className="flex-container">
        <div className="box grow1 yel">
          <h4>KDO PŘIPRAVUJE</h4>
          <p>BIO</p>
        </div>
        <div className="box grow3 yel">
          <h4>POZNÁMKA LABORATOŘE</h4>
          <p>{vv.labInfo}</p>
        </div>
      
      </div>
      
      <div className="flex-container">
        <div className="box grow1 posledni green">
          <h4>PROVÁDÍ SE</h4>
          <p>{vv.provadiSe}</p>
        </div>
        <div className="box grow1 posledni green">
          <h4>ODEZVA</h4>
          <p>{vv.odezvaDo}</p>
        </div>
        <div className="box grow1 posledni green">
          <h4>METODIKA</h4>
          <p>{vv.metodika}</p>
        </div>
        <div className="box grow1 posledni green">
          <h4>JEDNOTKA</h4>
          <p>{vv.jednotka}</p>
        </div>
        <div className="box grow1 posledni green">
          <h4>ODBORNOST</h4>
          <p>{vv.odbornost}</p>
        </div>
        
      </div>
      
    
    </div>
  
    )
  };

   
};

//pro zprehledneni mapStateToProps, funkce vezme vysetreni z redux state (coz je array s objekty o vysetrenich) a vrati objekt jednoho konkretniho vysetreni, ktere je prave otevrene v komponente Detail...
const najdiVysetreniVeReduxPodleId = (state, props) => {
  return state.vysetreni.find((polozka) => polozka.id === props.match.params.id)
};

const mapStateToProps = (state,props) => {

  

    // do props pridame jeden objekt z array vysetreni, ktery odpovida id z props.match.params.id(tedy to vysetreni na ktere jsme kliknuli)
 
    return {
        vv: najdiVysetreniVeReduxPodleId(state, props), //vv= vybrane vysetreni, prohleda array vysetreni, kde jsou vsechny objekty a vrati ten objekt, ktery odpovida podle id
        uzivatel: state.auth.uid,
        prevPoznOdd:  najdiVysetreniVeReduxPodleId(state, props).poznamkaOdd //najde objekt v array podle id a jen jeho poznamkuOdd, stejne jako vysetreni[x].poznamkaOdd
        
        };
};

const mapDispatchToProps = (dispatch) => {
  return {
    provedDispatchPridatPoznamkuOdd: (id, poznamkaObjekt) => dispatch(startPridatPoznamkuOdd(id, poznamkaObjekt))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(DetailVysetreniPage);