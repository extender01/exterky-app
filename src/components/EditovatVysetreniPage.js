import React from "react";
import  { connect } from "react-redux";
import VysetreniForm from "./VysetreniForm";
import { editovatVysetreni, odstranitVysetreni } from "../actions/vysetreniActions";


export class EditovatVysetreniPage extends React.Component {
    priEditaci = (objektUdajuZFormulare) => {
        this.props.editovatVysetreni(this.props.vybraneVysetreni.id, objektUdajuZFormulare);
        this.props.history.push("/");
    };


    priOdstraneni = () => {
        this.props.odstranitVysetreni(this.props.vybraneVysetreni.id);
        this.props.history.push("/");
    };


    render() {
        return (
            <div>
            Edituj lab vysetreni s id:  {this.props.vybraneVysetreni.id}
        
                <VysetreniForm 
                    vysetreniProEditaci={this.props.vybraneVysetreni}
                    provedSubmitDoStore={this.priEditaci}
                />
                <button onClick={this.priOdstraneni}
                >ODstranit</button>
            </div>
        );
    };


};




const mapStateToProps = (state,props) => {
    // do props pridame jeden objekt z array vysetreni, ktery odpovida id z props.match.params.id(tedy to vysetreni na ktere jsme kliknuli)
    //takze jsme to konkretni vysetreni vytahli z redux store jako vybraneVysetreni a do VysetreniForm ho posleme v props pod jmenem vysetreniProEditaci
    return {
        vybraneVysetreni: state.vysetreni.find((polozka) => polozka.id === props.match.params.id)
        };
    };


const mapDispatchToProps = (dispatch) => ({
    editovatVysetreni: (idGen, vysetreniGen) => dispatch(editovatVysetreni(idGen, vysetreniGen)),
    odstranitVysetreni: (objektsIdPropertyGen) => dispatch(odstranitVysetreni(objektsIdPropertyGen))
});


//mapStateToProps NASBIRA POTREBNE INFROMACE Z REDUX STORE (STATE OBJEKT) A VLOZI JE DO PROPS OBJEKTU POUZITELNEHO V KOMPONENTE
//DRUHY ARGUMENT PROPS JE UZ EXISTUJICI OBJEKT PROPS, KE KTEREMU JEN NECO PRIDAME (TADY UZ JE V PROPS INFORMACE O ID Z PROPS.MATCH.PARAMS.ID A MY K TOMU OBJEKTU PROPS PRIDAME DALSI INFORMACE Z REDUX STORE)
export default connect(mapStateToProps, mapDispatchToProps)(EditovatVysetreniPage);