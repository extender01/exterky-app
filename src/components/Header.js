import React from "react";
import { NavLink } from "react-router-dom";


const Header = () => (
    <header>
        <h1>Externi lab vysetreni</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Domu   </NavLink>
        <NavLink to="/create" activeClassName="is-active">Pridej   </NavLink>
        <NavLink to="/help" activeClassName="is-active">Pomoc   </NavLink>
        

    </header>
);

export default Header;