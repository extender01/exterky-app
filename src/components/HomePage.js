import React from "react";


const HomePage = () => (
    <div>
        <h1>Homepage</h1>
        <p>Donec vitae arcu. Vestibulum fermentum tortor id mi. Integer pellentesque quam vel velit. Sed vel lectus. Donec odio tempus molestie, porttitor ut, iaculis quis, sem. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Fusce wisi. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Aliquam erat volutpat. Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nulla non lectus sed nisl molestie malesuada. Maecenas aliquet accumsan leo. Nullam sapien sem, ornare ac, nonummy non, lobortis a enim. Nunc dapibus tortor vel mi dapibus sollicitudin.
        </p>
        </div>
);

export default HomePage;