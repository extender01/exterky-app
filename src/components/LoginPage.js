import React from "react";
import { connect } from "react-redux";
import { startPrihlasit } from "../actions/authActions";

export class LoginPage extends React.Component {

    constructor(props) {
        super(props);
    };

    provedSubmit = (e) => {
        e.preventDefault();
        const login = e.target.children[0].value;
        const password = e.target.children[1].value;
        this.props.startPrihlasit(login, password);
        console.log("v provedSubmit", login, password);
        
    }
   
   
    render() {

        return (
            <div className="login">
                <form name="formular" className="login__form" onSubmit={this.provedSubmit}>
                    <input type="text" placeholder="Uzivatelske jmeno"  className="login__input" name="id" />
                    <input type="password" placeholder="Heslo" className="login__input" name="password" />
                    <button className="login__button" >LOGIN</button>

                </form>
            
            </div>
        )
    }   


}





const mapDispatchToProps = (dispatch) => ({
    startPrihlasit: (login, password) => dispatch(startPrihlasit(login, password)) //provede prihlaseni do firebase
});


export default connect(undefined, mapDispatchToProps)(LoginPage);

