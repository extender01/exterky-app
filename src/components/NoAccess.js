import React from "react";
import { Link } from "react-router-dom";



const NoAccess = () => (
    <div>
        Nemas pristup, prihlas se - <Link to="/login" replace={true} >Login</Link>
    </div>
);

export default NoAccess;