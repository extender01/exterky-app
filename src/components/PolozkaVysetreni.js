import React from "react";
import { connect } from "react-redux";
import { odstranitVysetreni } from "../actions/vysetreniActions";
import { Link } from "react-router-dom";

const PolozkaVysetreni = (props) => (
    <div>
        <p>===========================</p>
        <Link to={"/edit/" + props.id} ><h3>{props.nazev}</h3> </Link>

        <p>{props.kam} / {props.odber}</p>

        

    </div>
);

export default PolozkaVysetreni;