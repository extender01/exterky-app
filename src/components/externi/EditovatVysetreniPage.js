import React from "react";
import  { connect } from "react-redux";
import VysetreniForm from "./VysetreniForm";
import { startEditovatVysetreni, startOdstranitVysetreni } from "../../actions/vysetreniActions";


export class EditovatVysetreniPage extends React.Component {
    priEditaci = (objektUdajuZFormulare) => {
        this.props.startEditovatVysetreni(this.props.vybraneVysetreni.id, objektUdajuZFormulare);
        this.props.history.push("/");
    };


    priOdstraneni = () => {
        this.props.startOdstranitVysetreni(this.props.vybraneVysetreni.id);
        this.props.history.push("/");
    };


    render() {
        return (
            <div>
            <p>Edituj lab vysetreni s id:  {this.props.vybraneVysetreni.id} a nazvem: {this.props.vybraneVysetreni.nazev} </p>
        
                <VysetreniForm 
                    vysetreniProEditaci={this.props.vybraneVysetreni}
                    provedSubmitDoStore={this.priEditaci}
                    uzivatel={this.props.uzivatel}
                />
                <button onClick={this.priOdstraneni}
                >ODstranit</button>
            </div>
        );
    };


};




const mapStateToProps = (state,props) => {
    // do props pridame jeden objekt z array vysetreni, ktery odpovida id z props.match.params.id(tedy to vysetreni na ktere jsme kliknuli)
    //takze jsme to konkretni vysetreni vytahli z redux store jako vybraneVysetreni a do VysetreniForm ho posleme v props pod jmenem vysetreniProEditaci
    return {
        vybraneVysetreni: state.vysetreni.find((polozka) => polozka.id === props.match.params.id),
        uzivatel: state.auth.uid
        };
};


const mapDispatchToProps = (dispatch) => ({
    startEditovatVysetreni: (idGen, vysetreniGen) => dispatch(startEditovatVysetreni(idGen, vysetreniGen)),
    startOdstranitVysetreni: (objektsIdPropertyGen) => dispatch(startOdstranitVysetreni(objektsIdPropertyGen))
});


//mapStateToProps NASBIRA POTREBNE INFROMACE Z REDUX STORE (STATE OBJEKT) A VLOZI JE DO PROPS OBJEKTU POUZITELNEHO V KOMPONENTE
//DRUHY ARGUMENT PROPS JE UZ EXISTUJICI OBJEKT PROPS, KE KTEREMU JEN NECO PRIDAME (TADY UZ JE V PROPS INFORMACE O ID Z PROPS.MATCH.PARAMS.ID A MY K TOMU OBJEKTU PROPS PRIDAME DALSI INFORMACE Z REDUX STORE)
export default connect(mapStateToProps, mapDispatchToProps)(EditovatVysetreniPage);