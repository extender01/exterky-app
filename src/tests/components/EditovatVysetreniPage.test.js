import React from "react";
import { shallow } from "enzyme";
import { EditovatVysetreniPage } from "../../components/EditovatVysetreniPage";
import modelovaData from "../fixtures/vysetreniFixtures";



let editovatVysetreni, odstranitVysetreni, history, vybraneVysetreni, wrapper;

beforeEach(() => {
    editovatVysetreni = jest.fn();
    odstranitVysetreni = jest.fn();
    history = {push: jest.fn()};
    wrapper = shallow(
        //predavane funkce ve forme props simuluji jejich volani z expenseForm protoze tam jsou taky v props pres MapDispatchToProps
        <EditovatVysetreniPage
            editovatVysetreni={editovatVysetreni}
            odstranitVysetreni={odstranitVysetreni}
            history={history}
            vybraneVysetreni={modelovaData[1]}
        />
    );
});


test("mel by vyrenderovat EditovatVysetreniPage", () => {
    expect(wrapper).toMatchSnapshot();
});


test("mel by zvladnout edit vysetreni", () => {
    wrapper.find("VysetreniForm").prop("provedSubmitDoStore")(modelovaData[1]);
    expect(history.push).toHaveBeenLastCalledWith("/");
    expect(editovatVysetreni).toHaveBeenLastCalledWith(modelovaData[1].id, modelovaData[1]);
});

test("mel by zvladnout edit vysetreni", () => {
    wrapper.find("button").simulate("click");
    expect(history.push).toHaveBeenLastCalledWith("/");
    expect(odstranitVysetreni).toHaveBeenLastCalledWith(modelovaData[1].id);
});




