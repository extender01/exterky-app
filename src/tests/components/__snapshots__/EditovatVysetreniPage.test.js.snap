// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mel by vyrenderovat EditovatVysetreniPage 1`] = `
<div>
  Edituj lab vysetreni s id:  
  2
  <VysetreniForm
    provedSubmitDoStore={[Function]}
    vysetreniProEditaci={
      Object {
        "id": 2,
        "kam": "Novy jicin",
        "nazev": "cystatin C",
        "odber": Array [
          "plasma",
        ],
        "poznamka": "rychle",
        "poznamkaOdd": "posilat rano",
        "preanal": "standard",
        "synonyma": Array [
          "hem vychytavac",
        ],
        "upravenoKdy": 100,
      }
    }
  />
  <button
    onClick={[Function]}
  >
    ODstranit
  </button>
</div>
`;
